Install.bat

```diff
@@ -0,0 +1,199 @@
+@echo off
+setlocal enabledelayedexpansion
+set VENV_DIR=.venv
+set "requirements_file=requirements.txt"
+:main
+set "wrong_python=0"
+cls
+if exist "%VENV_DIR%" (
+    call:cecho red "Previous virtual environment detected. May cause issues if not deleted."
+    echo [33m
+    set /p "choice=Would you like to remove this virtual environment? [Y/N]:"
+    echo [0m
+    set "choice=!choice:~0,1!"
+    set "choice=!choice:~0,1!"
+    if /i "!choice!"=="Y" (
+         rmdir /s /q %VENV_DIR%
+    ) else if /i "!choice!"=="N" (
+        call:cecho red "You have chosen not to remove the current virtual environment. Please note this may cause issues with the installation."
+    ) else (
+        call:cecho red "Invalid choice. Please enter Y or N."
+        goto :again
+    )
+)
+cls
+python --version >nul 2>&1
+if %errorlevel% EQU 0 (
+    :: Find the location of the Python executable
+    for /f %%i in ('where python') do (
+        set "python_path_check=%%i"
+        echo !python_path_check!
+        !python_path_check! --version 2>NUL | findstr /R "3.10" > NUL
+        if %errorlevel% EQU 0 (
+            echo python_path_check | findstr /R "Python310" > NUL
+            if %errorlevel% EQU 0 (
+                set "standalone=!python_path_check!"
+            )
+            echo python_path_check | findstr /R "WindowsApps" > NUL
+            if %errorlevel% EQU 0 (
+                set "WindowsApps=!python_path_check!"
+            )
+        ) else (
+            set "wrong_python=1"
+        )
+    )
+    if NOT "!standalone!"=="" (
+        echo Using standalone Python
+        set "python_path=!standalone!"
+        goto python310
+    ) else if NOT "!WindowsApps!"=="" (
+        echo Using WindowsStore Python
+        set "python_path=!WindowsApps!"
+        goto python310
+    ) 
+    if !wrong_python! EQU 1 (
+        goto wrong_python_display
+    )
+) else (
+    goto wrong_python_display
+)
+:python310
+call:cecho green "Python 3.10.x is installed and running."
+:activate
+if exist "%VENV_DIR%" (  
+    echo Activating virtual environment...
+    call "%VENV_DIR%\Scripts\activate.bat"
+) else (
+    echo Creating virtual environment...
+    !python_path! -m venv %VENV_DIR%
+    goto activate
+)
+:: Upgrading pip  
+python -m pip install --upgrade pip >NUL
+if %errorlevel% equ 0 (
+    call:cecho green "Pip is up-to-date."
+) else (
+   call:cecho green "Pip has been upgraded to the latest version."
+)
+:: Check if requirements.txt exists
+if not exist %requirements_file% (
```

Comment:
How can this not exist? Like the user deleted it for some reason??

Reply:
that's a fair point. I guess it is kind of just a useless piece of code, isn't it?